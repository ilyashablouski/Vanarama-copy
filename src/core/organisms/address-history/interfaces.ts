import { IBaseProps } from '../../interfaces/base';
import { IDropdown } from '../../interfaces/dropdown';
import { IAddress } from '../../interfaces/address';

interface IPropertyStatuses extends IDropdown {}

export interface IAddressHistoryObject {
  address: IAddress;
  propertyStatus: string;
  month: number;
  year: number;
}

export interface IAddressHistoryProps extends IBaseProps {
  name: string;
  minYearsOfHistory: number;
  lookBackYears: number;
  propertyStatuses: IPropertyStatuses;
  addressHistory?: IAddressHistoryObject[];
  setValue?: Function; // Callback.
}

export interface IAddressHistoryState {
  addressHistory?: IAddressHistoryObject[];
  completed: boolean;
  allValid: boolean;
}
