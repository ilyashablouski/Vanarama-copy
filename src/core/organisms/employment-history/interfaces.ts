import { IBaseProps } from '../../interfaces/base';
import { IDropdown } from '../../interfaces/dropdown';
import { IAddress } from '../../interfaces/address';

interface IEmploymentStatuses extends IDropdown {}

export interface IEmploymentHistoryObjectSlim {
  employmentStatus: string;
  month: number | null;
  year: number | null;
}

export interface IEmploymentHistoryObjectFull
  extends IEmploymentHistoryObjectSlim {
  contractType: string;
  jobTitle: string;
  companyName: string;
  phoneNumber: string;
  address: IAddress | null;
  income: string;
}

export type TEmploymentHistory =
  | IEmploymentHistoryObjectSlim
  | IEmploymentHistoryObjectFull;

export interface IEmploymentHistoryProps extends IBaseProps {
  minYearsOfHistory: number;
  lookBackYears: number;
  employmentStatuses: IEmploymentStatuses;
  employmentHistory?: TEmploymentHistory[];
  setValue?: Function; // Callback.
}

export interface IEmploymentHistoryState {
  employmentHistory: TEmploymentHistory[];
  completed: boolean;
  allValid: boolean;
}
