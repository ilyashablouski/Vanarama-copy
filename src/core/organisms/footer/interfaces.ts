import { IBaseProps } from '../../interfaces/base';
import { ILink } from '../../interfaces/link';

export interface IFooterProps extends IBaseProps {
  emailAddress?: string;
  phoneNumber?: string;
  openingTimes?: IOpeningTimes[];
  headingZero?: string;
  headingOne?: string;
  headingTwo?: string;
  headingThree?: string;
  linkGroupOne?: ILink[];
  linkGroupTwo?: ILink[];
  linkGroupThree?: ILink[];
  legalText?: string;
}

export interface IOpeningTimes {
  day: string;
  time: string;
}
