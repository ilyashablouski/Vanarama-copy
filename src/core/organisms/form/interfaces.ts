import { FormHTMLAttributes } from 'react';
import { IBaseProps } from '../../interfaces/base';

export interface IFormProps
  extends IBaseProps,
    FormHTMLAttributes<HTMLFormElement> {
  children: React.ReactNode;
  /**
   * Whether there a validation errors in this form
   */
  invalid?: boolean;
  /**
   * The function to call when the form is submitted
   */
  onSubmit?: (event: React.FormEvent<HTMLFormElement>) => void;
}

export interface IFormErrorProps extends IBaseProps {
  children: React.ReactNode;
}
