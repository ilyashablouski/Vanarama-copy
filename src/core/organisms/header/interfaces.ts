import { IBaseProps } from '../../interfaces/base';
import { ILink } from '../../interfaces/link';

export interface IHeaderProps extends IBaseProps {
  topBarLinks?: ILink[];
  loginLink?: ILink;
  phoneNumber?: string;
  showIvan?: boolean;
  message?: string;
}
