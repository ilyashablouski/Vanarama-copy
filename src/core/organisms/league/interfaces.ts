import { IBaseProps } from '../../interfaces/base';

export interface ILeagueProps extends IBaseProps {
  link: string;
  altText?: string;
}
