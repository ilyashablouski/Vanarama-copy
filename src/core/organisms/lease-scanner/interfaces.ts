import { IBaseProps } from '../../interfaces/base';

export interface ILeaseScannerProps extends IBaseProps {
  price: number;
  headingText: string;
  withCheckBox?: boolean;
  leasingProviders: string[];
  /**
   * Update price request status
   */
  startLoading: boolean;
  orderNowClick: () => void;
  nextBestPrice?: string;
  classNameHeading?: string;
  priceLabel?: string;
  endAnimation: () => void;
  requestCallBack?: () => void;
}
