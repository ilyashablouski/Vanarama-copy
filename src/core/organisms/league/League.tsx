import React, { FC, memo } from 'react';

import Link from '../../atoms/link';
import Text from '../../atoms/text';
import Heading from '../../atoms/heading';
import Image from '../../atoms/image';

import { ILeagueProps } from './interfaces';

const League: FC<ILeagueProps> = memo(props => {
  const { clickReadMore, altText, link } = props;

  return (
    <>
      <div>
        <Image
          plain
          src="https://www.vanarama.com/Assets/images-optimised/home/national-league.png"
          size="expand"
          alt={altText}
        />
      </div>
      <div>
        <Heading color="black" size="large" tag="span">
          The Vanarama National League
        </Heading>
        <Text color="darker" size="lead" tag="span">
          Read About Our Sponsorship{' '}
          <Link
            color="teal"
            onClick={() => {
              clickReadMore();
            }}
            href={link}
          >
            Here
          </Link>
        </Text>
        <Image
          plain
          src="https://www.vanarama.com/Assets/images-optimised/home/league-cup.png"
          size="expand"
          alt={altText}
        />
      </div>
    </>
  );
});

League.displayName = 'League';

export default React.memo(League);
