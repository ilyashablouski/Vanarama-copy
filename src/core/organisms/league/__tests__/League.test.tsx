import React from 'react';
import renderer from 'react-test-renderer';

import League from '..';
import { ILeagueProps } from '../interfaces';

const optionalProps = {
  altText: 'image1',
  link: '',
};

function getComponent(props?: ILeagueProps) {
  return renderer.create(<League {...props} link="" />).toJSON();
}

describe('<League />', () => {
  it('renders correctly', () => {
    const tree = getComponent();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with optional props', () => {
    const tree = getComponent(optionalProps);
    expect(tree).toMatchSnapshot();
  });
});
