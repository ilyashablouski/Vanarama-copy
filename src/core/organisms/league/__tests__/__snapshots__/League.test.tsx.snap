// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<League /> renders correctly 1`] = `
Array [
  <div>
    <div
      className="image -expand -plain"
    >
      <img
        className="image--native"
        loading="eager"
        sizes="(min-width:3200px) 800px, 1200px"
        src="https://vanarama-nonprod.com/cdn-cgi/image/fit=scale-down,height=1920,width=1920,quality=75,sharpness=0,metadata=none/https://dev.vanarama-nonprod.com/Assets/images-optimised/home/national-league.png"
        srcSet="https://vanarama-nonprod.com/cdn-cgi/image/fit=scale-down,height=320,width=320,quality=60,sharpness=0,metadata=none/https://dev.vanarama-nonprod.com/Assets/images-optimised/home/national-league.png 320w, https://vanarama-nonprod.com/cdn-cgi/image/fit=scale-down,height=800,width=800,quality=70,sharpness=0,metadata=none/https://dev.vanarama-nonprod.com/Assets/images-optimised/home/national-league.png 800w, https://vanarama-nonprod.com/cdn-cgi/image/fit=scale-down,height=1200,width=1200,quality=75,sharpness=0,metadata=none/https://dev.vanarama-nonprod.com/Assets/images-optimised/home/national-league.png 1200w"
      />
    </div>
  </div>,
  <div>
    <span
      className="heading -black -large"
    >
      The Vanarama National League
    </span>
    <span
      className="text -lead -darker"
    >
      Read About Our Sponsorship
       
      <a
        className="link -teal -regular -plain"
        href=""
        onClick={[Function]}
      >
        Here
      </a>
    </span>
    <div
      className="image -expand -plain"
    >
      <img
        className="image--native"
        loading="eager"
        sizes="(min-width:3200px) 800px, 1200px"
        src="https://vanarama-nonprod.com/cdn-cgi/image/fit=scale-down,height=1920,width=1920,quality=75,sharpness=0,metadata=none/https://dev.vanarama-nonprod.com/Assets/images-optimised/home/league-cup.png"
        srcSet="https://vanarama-nonprod.com/cdn-cgi/image/fit=scale-down,height=320,width=320,quality=60,sharpness=0,metadata=none/https://dev.vanarama-nonprod.com/Assets/images-optimised/home/league-cup.png 320w, https://vanarama-nonprod.com/cdn-cgi/image/fit=scale-down,height=800,width=800,quality=70,sharpness=0,metadata=none/https://dev.vanarama-nonprod.com/Assets/images-optimised/home/league-cup.png 800w, https://vanarama-nonprod.com/cdn-cgi/image/fit=scale-down,height=1200,width=1200,quality=75,sharpness=0,metadata=none/https://dev.vanarama-nonprod.com/Assets/images-optimised/home/league-cup.png 1200w"
      />
    </div>
  </div>,
]
`;

exports[`<League /> renders correctly with optional props 1`] = `
Array [
  <div>
    <div
      className="image -expand -plain"
    >
      <img
        alt="image1"
        className="image--native"
        loading="eager"
        sizes="(min-width:3200px) 800px, 1200px"
        src="https://vanarama-nonprod.com/cdn-cgi/image/fit=scale-down,height=1920,width=1920,quality=75,sharpness=0,metadata=none/https://dev.vanarama-nonprod.com/Assets/images-optimised/home/national-league.png"
        srcSet="https://vanarama-nonprod.com/cdn-cgi/image/fit=scale-down,height=320,width=320,quality=60,sharpness=0,metadata=none/https://dev.vanarama-nonprod.com/Assets/images-optimised/home/national-league.png 320w, https://vanarama-nonprod.com/cdn-cgi/image/fit=scale-down,height=800,width=800,quality=70,sharpness=0,metadata=none/https://dev.vanarama-nonprod.com/Assets/images-optimised/home/national-league.png 800w, https://vanarama-nonprod.com/cdn-cgi/image/fit=scale-down,height=1200,width=1200,quality=75,sharpness=0,metadata=none/https://dev.vanarama-nonprod.com/Assets/images-optimised/home/national-league.png 1200w"
      />
    </div>
  </div>,
  <div>
    <span
      className="heading -black -large"
    >
      The Vanarama National League
    </span>
    <span
      className="text -lead -darker"
    >
      Read About Our Sponsorship
       
      <a
        className="link -teal -regular -plain"
        href=""
        onClick={[Function]}
      >
        Here
      </a>
    </span>
    <div
      className="image -expand -plain"
    >
      <img
        alt="image1"
        className="image--native"
        loading="eager"
        sizes="(min-width:3200px) 800px, 1200px"
        src="https://vanarama-nonprod.com/cdn-cgi/image/fit=scale-down,height=1920,width=1920,quality=75,sharpness=0,metadata=none/https://dev.vanarama-nonprod.com/Assets/images-optimised/home/league-cup.png"
        srcSet="https://vanarama-nonprod.com/cdn-cgi/image/fit=scale-down,height=320,width=320,quality=60,sharpness=0,metadata=none/https://dev.vanarama-nonprod.com/Assets/images-optimised/home/league-cup.png 320w, https://vanarama-nonprod.com/cdn-cgi/image/fit=scale-down,height=800,width=800,quality=70,sharpness=0,metadata=none/https://dev.vanarama-nonprod.com/Assets/images-optimised/home/league-cup.png 800w, https://vanarama-nonprod.com/cdn-cgi/image/fit=scale-down,height=1200,width=1200,quality=75,sharpness=0,metadata=none/https://dev.vanarama-nonprod.com/Assets/images-optimised/home/league-cup.png 1200w"
      />
    </div>
  </div>,
]
`;
